source/boundary_temperature/box.cc

```diff
@@ -149,6 +142,13 @@ namespace aspect
               default:
                 Assert (false, ExcNotImplemented());
             }
+
+          // verify that the geometry is a box since only for this geometry
+          // do we know for sure what boundary indicators it uses and what they mean
+          AssertThrow (Plugins::plugin_type_matches<const GeometryModel::Box<dim>>(this->get_geometry_model()),
```

Comment:
It might be nice to make sure that this check is in the same place for all plugins. I think the init function is the best place to put this since it is called earlier. The downside of that would be that extra functions need to be added to the plugins. What do you think?

Reply:
parse_parameters() is also okay with me as all objects are already constructed. Both are better than during the actual usage.